mantidproject/mantid
# Indirect - Transmission - Container and Sample spectra plotted incorrectly in plot preview

In Indirect Transmission, the container and sample spectra were swapped in the plot preview. This was due to using indexing of a grouped workspace; assuming container, sample and transmission workspaces would always have the same index in a grouped workspace - this is not necessarily true.

Fixed this issue by using the names of the workspaces instead. Also changed green to blue (as suggested by Spencer) - this is done only in the plot preview.

**To test:**
1. Navigate to Interfaces > Indirect > Data Reduction.
2. Navigate to the Transmission tab.
3. Enter a sample file into the 'Sample' field (e.g. 26180 - with Use Data Archive enabled).
4. Enter a container file into the 'Background' field (e.g. 26174 - with Use Data Archive enabled).
5. Keep the plot preview open and plot the spectra of the grouped workspace in the ADS using 'Plot Spectrum'.
6. Ensure both of these plots are equivalent.



Fixes #18749. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
